newa/__init__.py

```diff
@@ -968,3 +1014,194 @@ def drop_obsoleted_issue(self, issue: Issue, obsoleted_by: Issue) -> None:
                                              transition=self.transitions["dropped"][0])
         except jira.JIRAError as e:
             raise Exception(f"Cannot close issue {issue}!") from e
+
+#
+# ReportPortal communication
+#
+
+
+@define
+class ReportPortal:
+
+    token: str
+    url: str
+    project: str
+
+    def get_launch_url(self, launch_id: str) -> str:
+        return f"{self.url}/ui/#{self.project}/launches/all/{launch_id}"
```

Comment:
To stay on the safe side, I strongly suggest using `urllib.parse.quote` (https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) when injecting our strings into URLs. Our strings may contain characters that must not appear in URL - space, maybe even diacritics, or characters like `?`. I suppose most of the time, this will not be the case, but it's a good and safe habit.

```python
from urllib.parse import quote as Q

return f"{self.url}/ui/#{Q(self.project)}/launches/all/{Q(launch_id)}"
```

It should be applied to pretty much every component we use to construct URL, except the schema (`urllib.parse.quote('http://foo.bar.com')` => `'http%3A//foo.bar.com'`).

Reply:
fixed